lib/Doctrine/Common/Reflection/TypedNoDefaultReflectionProperty.php

```diff
@@ -31,7 +31,7 @@ public function getValue($object = null)
      */
     public function setValue($object, $value = null)
     {
-        if ($value === null) {
+        if ($value === null && ! $this->getType()->allowsNull()) {
```

Comment:
In case of a non-typed property, `ReflectionProperty::getType()` will return `null`.

Reply:
but its by definition a typed property here.

Reply:
Okay, in that case this shouldn't be an issue if the class, that has to be used for a property, is determined elsewhere and will always result in `TypedNoDefaultReflectionProperty` for PHP 7.4 when typed properties are in use.

Reply:
@SenseException i am not sure anymore, so i added a `hasType()` check inbetween.

Reply:
Then I'd like to see a test where no property type is given.